Add version to test tool

normj opened the pull request:
*Description of changes:*
Add a `--version` switch that if set prints the tool's version as a JSON document. This allows other tooling like the Aspire integration to shell out to the command and programmatically check the installed version. JSON document was chosen to allow the possibility of expanding on the info returned.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

GarrettBeatty commented on Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Commands/RunCommand.cs:
```diff
@@ -28,6 +29,12 @@ public override async Task<int> ExecuteAsync(CommandContext context, RunCommandS
     {
         try
         {
+            if (settings.PrintVersionInfo)
```
can we add a unit test to https://github.com/aws/aws-lambda-dotnet/blob/8c1705f523c9d75db87bc326f5ee3d5ec6de39be/Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.UnitTests/Commands/RunCommandTests.cs#L17? 

Also as a side note, do we want to enforce unit test coverage for future PRs as a check? What do we do for other projects that we have?

normj replied:
I was being reluctant adding a mocking layer around stdout so I could reliable unit test this. Turns out we are already mocking stdout with `IToolInteractiveService` so I added the unit test.

normj commented:
I renamed the `--version` switch to `--tool-info` because I though the name seemed more broad then return just the version number. I also added the `install-path` of the tooling. I though that might come in handy when we work on setting up launchsettings.json file.

philasmar commented on .autover/changes/efa829ed-59af-46eb-a5e5-ef471698c82d.json:
```diff
@@ -0,0 +1,11 @@
+      "Name": "Amazon.Lambda.TestTool",
+      "Type": "Patch",
+      "ChangelogMessages": [
+        "Add --version switch to allow integrators to check what version is installed"
```
you missed updating the changelog

philasmar reviewed (commented):
I'm ok with the change, but instead of us putting everything new thing in RunCommand. We should create another command for info. So the command would be `dotnet-lambda-test-tool info`. That way, we can potentially add further command line arguments to format the info we are looking for.

normj commented on Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Commands/Settings/RunCommandSettings.cs:
```diff
@@ -36,22 +36,6 @@ public sealed class RunCommandSettings : CommandSettings
     [Description("Disable auto launching the test tool's web interface in a browser.")]
     public bool NoLaunchWindow { get; set; }
 
-    /// <summary>
```
These were leftover from the old version of the tooling and are not used in the new version.
